core/cap-0065.md

```diff
@@ -0,0 +1,326 @@
+3. Update issuer semantics in the SAC so that a `transfer` involving the issuer will emit the semantically correct event (`mint` or `burn`).
+
+The added events will not be a protocol change because they won't be hashed into the ledger, but the Stellar Asset Contract changes will be.
+
+## Specification
+
+### XDR Changes
+
+This patch of XDR changes is based on the XDR files in commit `734bcccdbb6d1f7e794793ad3b8be51f3ba76f92` of stellar-xdr.
+```
+diff --git a/Stellar-contract.x b/Stellar-contract.x
+index 5113005..112e784 100644
+--- a/Stellar-contract.x
++++ b/Stellar-contract.x
+@@ -70,7 +70,10 @@ enum SCValType
+     // symbolic SCVals used as the key for ledger entries for a contract's
+     // instance and an address' nonce, respectively.
+     SCV_LEDGER_KEY_CONTRACT_INSTANCE = 20,
+-    SCV_LEDGER_KEY_NONCE = 21
++    SCV_LEDGER_KEY_NONCE = 21,
++
++    // Contains other address types that are not used within contracts
++    SCV_ADDRESS_EXTENDED = 22
+ };
+ 
+ enum SCErrorType
+@@ -190,6 +193,20 @@ case SC_ADDRESS_TYPE_CONTRACT:
+     Hash contractId;
+ };
+ 
++enum SCAddressExtendedType
++{
++    SC_ADDRESS_EXTENDED_TYPE_CLAIMABLE_BALANCE = 0,
++    SC_ADDRESS_EXTENDED_TYPE_LIQUIDITY_POOL = 1
++};
++
++union SCAddressExtended switch (SCAddressExtendedType type)
++{
++case SC_ADDRESS_EXTENDED_TYPE_CLAIMABLE_BALANCE:
++    ClaimableBalanceID claimableBalanceID;
++case SC_ADDRESS_EXTENDED_TYPE_LIQUIDITY_POOL:
++    PoolID liquidityPoolID;
++};
++
+ %struct SCVal;
+ %struct SCMapEntry;
+ 
+@@ -271,6 +288,9 @@ case SCV_LEDGER_KEY_NONCE:
+ 
+ case SCV_CONTRACT_INSTANCE:
+     SCContractInstance instance;
++
++case SCV_ADDRESS_EXTENDED:
++    SCAddressExtended extendedAddress;
+ };
+ 
+ struct SCMapEntry
+
+```
+
+## Semantics
+
+### Remove the admin from the SAC `mint` and `clawback` events
+
+The `mint` event will look like:
+```
+contract: asset, topics: ["mint", to:Address, sep0011_asset:String], data: amount:i128
+```
+
+The `clawback` event will look like:
+```
+contract: asset, topics: ["clawback", from:Address, sep0011_asset:String], data: amount:i128
+```
+
+### Emit the semantically correct event for a Stellar Asset Contract `transfer` when the issuer is involved 
+
+At the moment, if the issuer is the sender in a Stellar Asset Contract `transfer`, the asset will be minted. If the issuer is the recipient, the asset will be burned. The event emitted in both scenarios, however, is the `transfer` event. This CAP changes that behavior to instead emit the `mint`/`burn` event.
+
+### New Events
+This section will go over the semantics of how the additional `transfer` events are emitted for each operation, as well as the `fee` event emitted for the fee paid by the source account. These events will be emitted through `TransactionMeta`, and will not be hashed into the ledger. Note that
+the `contract` field for these events corresponds to the Stellar Asset Contract address for the respective asset.
```

Comment:
It may be worth noting that the SAC is not required to be deployed and that this will result in events being published for a contract address that is reserved, but not deployed.